src/components/SingleArticle.jsx

```diff
@@ -16,6 +20,23 @@ const SingleArticle = () => {
         });
       }, [article_id]);
 
+      const handleVote = (voteType) => {
+        if (hasVoted) {
+            setVoteError("Ooops! Looks like you can only vote once!");
+        } else {
+            const newVotesCount = voteType === 'up' ? article.votes + 1 : article.votes - 1;
+
```

Comment:
Before doing the patch request, you should optimistically render the new votes in the page using a state for that purpose.